testruby.c

```diff
@@ -1,24 +1,51 @@
 #include <stdio.h>
 #include <stdlib.h>
-int i, M, N, x, y, j, z, var, red, st1, st2, st, red1, red2, l, d;
-char c = 'A', b, pamti, boja;
-char **tablica;
-void regija(int x, int y, char boja, char pamti)
+#include <stdbool.h>
+int M, N;
+char **matrix;
+bool isPixelValid(int x, int y)
 {
-    if (tablica[x][y] == boja)
+    if (x < 0 || x > N - 1 || y < 0 || y > M - 1)
+        return false;
+    else
+        return true;
+}
+char getPixelColor(int x, int y)
+{
+    if (isPixelValid(x, y) == true)
+        return matrix[x][y];
+        else return '\0';
+}
+char setPixelColor(int x, int y, char color)
+{
+    if (isPixelValid(x, y) == true)
+    {
+        matrix[x][y] = color;
+        return color;
+    }
+    else
+        return '\0';

```

Comment:
```suggestion
    if (!isPixelValid(x, y)) { return '\0'; }
    return matrix[x][y] = color;
        

```

Reply:
actually IMHO it is better to return FALSE instead of '\0' so we can then use result of setColor function in an if statement condition. 

also this is an interesting article, when returning \0 iit is really easy to make a mistake of mixing " AND '

https://stackoverflow.com/questions/36950552/0-evaluates-false-0-evaluates-true/36950573